src/main/scala/analysis/solvers/FixPointSolver.scala

```diff
@@ -87,19 +89,33 @@ trait ListSetWorklist[N] extends Worklist[N]:
 
   def add(ns: Set[N]) = worklist ++= ns
 
-  def run(first: Set[N]) =
+  def run(first: Set[N], intra: Boolean) =
     worklist = new ListSet[N] ++ first
     while (worklist.nonEmpty) do
       val n = worklist.head;
       worklist = worklist.tail
-      process(n)
-
-  def monotonic_run(first: Set[N]) =
-    worklist = new ListSet[N] ++ first.collect{ case n: CfgFunctionEntryNode => n }
-    while (worklist.nonEmpty) do
-      val n = worklist.head;
-      worklist = worklist.tail
-      process(n)
+      process(n, intra)
+
+  def monotonic_run(first: Set[N], intra: Boolean) =
+    if (!intra) {
+      worklist =
+        new ListSet[N] ++ first.collect {
+          case n: CfgFunctionEntryNode if n.data.name == "main" => n
```

Comment:
Shouldn't be using "main" to find the initial procedure now that we have Program.mainProcedure